handler.go

```diff
@@ -319,10 +319,14 @@ func useCache(theURL *url.URL) bool {
 
 func buildHealthcheckJSONResponse(w http.ResponseWriter, healthResult fthealth.HealthResult) {
 	w.Header().Set("Content-Type", "application/json")
-	enc := json.NewEncoder(w)
-	err := enc.Encode(healthResult)
+	jData, err := json.Marshal(healthResult)
 	if err != nil {
-		panic("Couldn't encode health results to ResponseWriter.")
+		log.WithError(err).Error("Marshaling healthResult failed.")
+
```

Comment:
I would remove the empty line and (more importantly) add Error 500:

w.WriteHeader(http.StatusInternalServerError)
w.Write("{\"message": "Internal server error"}\")
return

Reply:
This is already addressed.